### PR TITLE
[sailfish-webview] Introduce webview test application. Contributes to JB#56037

### PR DESCRIPTION
**Pull request by paveltkv**

### CODE REVIEW

**Review by atatarov — APPROVED** [review]

**Review by okodron — APPROVED** [review]

File: examples/webview-test-app/main.qml [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+                    active = value
+                }
+
+                anchors.fill: parent
+                active: false
+                sourceComponent: webViewComponent
+
+                onItemChanged: {
+                    if (active) {
```
Comment (rainemak) [inline examples/webview-test-app/main.qml:226]:
```suggestion
                    if (item) {
```

File: examples/webview-test-app/main.qml [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+                        console.log("page loading: " + webPageAddress)
+                        item.load(webPageAddress)
+                    }
+                }
+            }
+
+            Rectangle {
+                id: loadStatus
```
Comment (rainemak) [inline examples/webview-test-app/main.qml:234]: Unused

File: examples/webview-test-app/main.qml [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+                color: "grey"
+                anchors.bottom: parent.bottom
+                anchors.left: parent.left
+                anchors.margins: 22
```
Comment (rainemak) [inline examples/webview-test-app/main.qml:239]: It's good habit to group anchors when possible. So like:

```
anchors {
    bottom: parent.bottom
    left: parent.left
    margins: 22
}
```

File: examples/webview-test-app/SpinBox.qml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    property real stepSize: 0.1
+
+    implicitHeight: Theme.itemSizeMedium
+    implicitWidth: 200
+
+    onValueChanged: {
+        value = Math.min( Math.max(minimumValue, value), maximumValue )
+    }
+
+    IconButton {
```
Comment (rainemak) [inline examples/webview-test-app/SpinBox.qml:19]: For this kind of a case Row work nicely. By using Row you don't need to fiddle with z-index and sibling anchors.

```
    Row {
        width: parent.width
        anchors.verticalCenter: parent.verticalCenter
        IconButton {
            id: buttonMinus
            width: height
            anchors.verticalCenter: textField.verticalCenter
            icon.source: "image://theme/icon-video-overlay-play"
            rotation: 180
            onClicked: root.value -= root.stepSize
        }

        TextField {
            id: textField

            inputMethodHints: Qt.ImhFormattedNumbersOnly
            placeholderText: label
            text: root.value.toFixed(2)
            horizontalAlignment: TextInput.AlignHCenter
            width: parent.width - 2*buttonMinus.width
            anchors.verticalCenter: parent.verticalCenter

            EnterKey.iconSource: "image://theme/icon-m-enter-next"
            EnterKey.onClicked: {
                focus = false;
                root.value = text;
            }

            onFocusChanged: {
                if (!focus) {
                    root.value = text;
                }
            }
        }

        IconButton {
            height: buttonMinus.height
            width: buttonMinus.height
            anchors.verticalCenter: textField.verticalCenter
            icon.source: "image://theme/icon-video-overlay-play"
            onClicked: root.value += root.stepSize
        }
    }
```

File: examples/webview-test-app/SpinBox.qml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    property real maximumValue: Infinity
+    property real value: 0.0
+    property real stepSize: 0.1
+
+    implicitHeight: Theme.itemSizeMedium
+    implicitWidth: 200
+
+    onValueChanged: {
+        value = Math.min( Math.max(minimumValue, value), maximumValue )
+    }
```
Comment (rainemak) [inline examples/webview-test-app/SpinBox.qml:17]: With one liner signal handlers it's not mandatory to use curly brackets. Following would be equally good:

```
onValueChanged: value = Math.min(Math.max(minimumValue, value), maximumValue)

File: examples/webview-test-app/main.qml [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+                    onLoadProgressChanged: {
+                        webViewPage.webViewLoadProgress = loadProgress
+                    }
+                }
+            }
+
+            Loader {
+                id: webViewLoader
+
+                function setWebViewActive(value) {
```
Comment (rainemak) [inline examples/webview-test-app/main.qml:207]: Is it not needed rather even a bit misleading. I'd rather just use ```webViewLoader.active``` where needed.

For logging purposes you could add

```
onActiveChanged: {
    if (value) {
        console.log("webview creation")
    } else {
        console.log("webview destruction")
    }
}
```

Changing a value of a property to the same that there was should not trigger change. That is the normal way how properties should be written in C++. This implies that...

```
webViewLoader.active = true
webViewLoader.active = true
```

Would log "webview creation" once given that previous active value was false.

**Review by rainemak — COMMENTED** [review]
Only minor remarks. Please give a look.

**Review by rainemak — APPROVED** [review]

**Pull request merged**